supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActor.scala

```diff
@@ -415,36 +477,83 @@ class TesAsyncBackendJobExecutionActor(override val standardParams: StandardAsyn
       }
     } yield statusWithLog
 
-  private def queryStatusAsync(handle: StandardAsyncPendingExecutionHandle): Future[TesRunStatus] =
-    makeRequest[MinimalTaskView](HttpRequest(uri = s"$tesEndpoint/${handle.pendingJob.jobId}?view=MINIMAL")) map {
-      response =>
-        val state = response.state
-        state match {
-          case s if s.contains("COMPLETE") =>
-            jobLogger.info(s"Job ${handle.pendingJob.jobId} is complete")
-            Complete
+  private def queryStatusAndCostData(handle: StandardAsyncPendingExecutionHandle,
+                                     fetchCostData: Boolean
+  ): Future[TesRunStatus] =
+    if (fetchCostData) {
+      makeRequest[Task](HttpRequest(uri = s"$tesEndpoint/${handle.pendingJob.jobId}?view=FULL")) map { response =>
+        val fetchedCostData = for {
+          responseLogs <- response.logs
+          startTime <- responseLogs.head.start_time
+          vmCost <- responseLogs.head.metadata.flatMap(_.get("vm_price_per_hour_usd"))
+        } yield (startTime, vmCost)
+
+        fetchedCostData match {
+          case Some(data) =>
+            val costData = Option(TesVmCostData(data._1, data._2))
+            val state = response.state
+            val metadata = Map(
+              CallMetadataKeys.TaskStartTime -> data._1,
+              CallMetadataKeys.VmCostUsd -> data._2
+            )
+            tellMetadata(metadata)
+            getTesStatus(state, costData, handle.pendingJob.jobId)
+          case None => getTesStatus(response.state, Option.empty, handle.pendingJob.jobId)
+        }
+      }
+    } else {
+      makeRequest[MinimalTaskView](HttpRequest(uri = s"$tesEndpoint/${handle.pendingJob.jobId}?view=MINIMAL")) map {
+        response =>
+          val state = response.state
+          getTesStatus(Option(state), Option.empty, handle.pendingJob.jobId)
```

Comment:
Rather than passing through empty cost data, should this be passing the cost data we already loaded?

Reply:
Right, I'm wondering if that's going to lead to the behavior we want. I might be missing something, but it looks like right now, if we already have all the cost data, when we poll we'll end up with a status without any cost data attached - which means we'll fetch it again next time we poll.